Performance Improvement | Catalog Page

arslanashraf7 opened the pull request:
#### Pre-Flight checklist

- [x] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Associated with #2067.
Parent Ticket: #2065 

#### What's this PR do?
Attempts to resolve/apply performance related suggestions from Lighthouse tool to optimize the overall performance of the catalog page

#### How should this be manually tested?
Check the analysis of `Lighthouse` and verify the increased performance then before.

#### Where should the reviewer start?
Checking analytics for `Lighthouse`

#### Any background context you want to provide?
(Optional)

gsidebo commented:
@arslanashraf7 Just FYI, I just opened a separate PR for the combined font import and async stylesheet loading on `base.html`: https://github.com/mitodl/mitxpro/pull/2080. There were a couple changes I was going to request on this PR and I'm eager to see the impact of that change, so I thought it would be best to just put it in a separate PR.

arslanashraf7 commented:
@gsidebo Sure, i agree with your idea. I can review the PR that you created so that we can take it to master to see it's impact.

However, There is an other PR https://github.com/mitodl/mitxpro/pull/2075 which is rebased on this one since most of the stuff was common and some some functionality defined in this PR was required in that. So, I thought i would get a review on that one and close this PR after applying the requested changes and merging that.

arslanashraf7 commented:
@gsidebo I've updated the other PR https://github.com/mitodl/mitxpro/pull/2075 (Which is basically rebased on this PR and needs review), I've removed the font optimization related changes from that PR in regard of https://github.com/mitodl/mitxpro/pull/2080.

arslanashraf7 commented:
Closing this as we have other PR https://github.com/mitodl/mitxpro/pull/2075 which will contain these changes as part of more general improvements rather the categorized by CMS Page.
